Extensions/ParticleSystem/ExtensionSubDeclaration1.cpp

```diff
@@ -122,6 +122,16 @@ void ExtensionSubDeclaration1(gd::ObjectMetadata& obj) {
       .AddParameter("object", _("Object"), "ParticleEmitter")
       .UseStandardRelationalOperatorParameters("number");
 
+  obj.AddCondition("ParticleType",
```

Comment:
I suggest we follow the term in the UI: `ParticleKind`.

I also suggest that instead of returning a number, we return a string ("Point", "Line" or "Quad") to be a bit more user friendly.

Finally, take a look at `AddExpressionAndCondition` that will allow you to create both a condition and an expression for it :) 
`obj.AddExpressionAndCondition("string", "ParticleKind", _("Kind of particle"), _("the kind of particle"), ...)`

It's fairly new but useful to reduce the number of declarations to do, and declare expressions.